src/quo2/components/avatars/user_avatar/view.cljs

```diff
@@ -1,72 +1,57 @@
 (ns quo2.components.avatars.user-avatar.view
-  (:require [clojure.string :as string]
-            [quo2.components.avatars.user-avatar.style :as style]
+  (:require [quo2.components.avatars.user-avatar.style :as style]
             [quo2.components.markdown.text :as text]
             [react-native.core :as rn]
             [react-native.fast-image :as fast-image]))
 
-(defn trim-whitespace [s] (string/join " " (string/split (string/trim s) #"\s+")))
-
-(defn- extract-initials
-  [full-name amount-initials]
-  (let [upper-case-first-letter (comp string/upper-case first)
-        names-list              (string/split (trim-whitespace full-name) " ")]
-    (if (= (first names-list) "")
-      ""
-      (->> names-list
-           (map upper-case-first-letter)
-           (take amount-initials)
-           (string/join)))))
-
-(defn initials-avatar
-  [{:keys [full-name size draw-ring? customization-color]}]
-  (let [font-size       (get-in style/sizes [size :font-size])
-        amount-initials (if (#{:xs :xxs :xxxs} size) 1 2)]
-    [rn/view
-     {:accessibility-label :initials-avatar
-      :style               (style/initials-avatar size draw-ring? customization-color)}
-     [text/text
-      {:style  style/initials-avatar-text
-       :size   font-size
-       :weight :semi-bold}
-      (extract-initials full-name amount-initials)]]))
-
-(def valid-ring-sizes #{:big :medium :small})
-
 (defn user-avatar
-  "If no `profile-picture` is given, draws the initials based on the `full-name` and
-  uses `ring-background` to display the ring behind the initials when given. Otherwise,
-  shows the `profile-picture` which already comes with the ring drawn."
-  [{:keys [full-name status-indicator? online? size profile-picture ring-background
-           customization-color static?]
-    :or   {status-indicator?   true
+  "Render user avatar with `profile-picture`
+  `profile-picture` should be one of {:uri profile-picture-uri} or {:fn profile-picture-fn}"
```

Comment:
docs added

> btw, why do we need that fn option? could you please explain a little more about it?

media server requires logic variables (public-key, key-uid, port ...) and style variables (size, color ...) to render the avatar
the fn here is to separate these variables so that some of them can be passed in in subs and some in user-avatar component

Reply:
Ok, thanks for the explanation